pkg/scheduler/testing/wrappers.go

```diff
@@ -285,6 +297,24 @@ func (p *PodWrapper) HostPort(port int32) *PodWrapper {
 	return p
 }
 
+// ContainerPort creates a container with ports valued `ports`,
+// and injects into the inner pod.
+func (p *PodWrapper) ContainerPort(ports []v1.ContainerPort) *PodWrapper {
+	p.Spec.Containers = append(p.Spec.Containers, v1.Container{
+		Ports: ports,
+	})
+	return p
+}
+
+// ContainerPortAtIndex sets the ports with value `ports` for the container at `index`
+func (p *PodWrapper) ContainerPortAtIndex(ports []v1.ContainerPort, index int) *PodWrapper {
```

Comment:
Not sure I like it as `index` may be out of bound.

Is there any case `ContainerPort` cannot fit?

Reply:
The only case where `ContainerPort` doesn't fit this `MakePod` at https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/internal/cache/cache_test.go#L1739

Since when we call `Req()` under `PodWrapper` it creates the container and we want to set the port for that container rather than create and append.

But I am doing something like this for now to have it fit using `ContainerPort`: https://github.com/kubernetes/kubernetes/pull/109684/files#diff-c93487173ed1e8915cdcdd46130ed723d517e191c35e817cfd3f76c913b94281R1707-R1711

Reply:
I feel like a ContainerWrapper is worthy, similar to NodeSelectorWrapper as a "second-level" wrapper, which is passed PodWrapper:

```go
st.MakePod().Containers(st.MakeContainer().Req().Port().Obj(), ...)
```

Reply:
Sure thing, created `ContainerWrapper` and refactored the existing `PodWrapper` code to leverage this. I agree that this does look cleaner.